profiles/contributed_browser_profiles.go

```diff
@@ -1 +1,144 @@
 package profiles
+
+import (
+	"github.com/bogdanfinn/fhttp/http2"
+	tls "github.com/bogdanfinn/utls"
+)
+
+var Firefox_120 = ClientProfile{
+	clientHelloId: tls.ClientHelloID{
+		Client:               "Firefox",
+		RandomExtensionOrder: false,
+		Version:              "120",
+		Seed:                 nil,
+		SpecFactory: func() (tls.ClientHelloSpec, error) {
+			return tls.ClientHelloSpec{
+				CipherSuites: []uint16{
+					tls.TLS_AES_128_GCM_SHA256,
+					tls.TLS_CHACHA20_POLY1305_SHA256,
+					tls.TLS_AES_256_GCM_SHA384,
+					tls.TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,
+					tls.TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,
+					tls.TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256,
+					tls.TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256,
+					tls.TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,
+					tls.TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,
+					tls.TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA,
+					tls.TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA,
+					tls.TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA,
+					tls.TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA,
+					tls.TLS_RSA_WITH_AES_128_GCM_SHA256,
+					tls.TLS_RSA_WITH_AES_256_GCM_SHA384,
+					tls.TLS_RSA_WITH_AES_128_CBC_SHA,
+					tls.TLS_RSA_WITH_AES_256_CBC_SHA,
+				},
+				CompressionMethods: []byte{
+					tls.CompressionNone,
+				},
+				Extensions: []tls.TLSExtension{
+					&tls.SNIExtension{},
+					&tls.ExtendedMasterSecretExtension{},
+					&tls.RenegotiationInfoExtension{Renegotiation: tls.RenegotiateOnceAsClient},
+					&tls.SupportedCurvesExtension{[]tls.CurveID{
+						tls.X25519,
+						tls.CurveP256,
+						tls.CurveP384,
+						tls.CurveP521,
+						tls.FAKEFFDHE2048,
+						tls.FAKEFFDHE3072,
+					}},
+					&tls.SupportedPointsExtension{SupportedPoints: []byte{
+						tls.PointFormatUncompressed,
+					}},
+
+					&tls.ALPNExtension{AlpnProtocols: []string{"h2", "http/1.1"}},
+					&tls.StatusRequestExtension{},
+					&tls.DelegatedCredentialsExtension{
+						SupportedSignatureAlgorithms: []tls.SignatureScheme{
+							tls.ECDSAWithP256AndSHA256,
+							tls.ECDSAWithP384AndSHA384,
+							tls.ECDSAWithP521AndSHA512,
+							tls.ECDSAWithSHA1,
+						},
+					},
+					&tls.KeyShareExtension{[]tls.KeyShare{
+						{Group: tls.X25519},
+						{Group: tls.CurveP256},
+					}},
+					&tls.SupportedVersionsExtension{[]uint16{
+						tls.VersionTLS13,
+						tls.VersionTLS12,
+					}},
+					&tls.SignatureAlgorithmsExtension{SupportedSignatureAlgorithms: []tls.SignatureScheme{
+						tls.ECDSAWithP256AndSHA256,
+						tls.ECDSAWithP384AndSHA384,
+						tls.ECDSAWithP521AndSHA512,
+						tls.PSSWithSHA256,
+						tls.PSSWithSHA384,
+						tls.PSSWithSHA512,
+						tls.PKCS1WithSHA256,
+						tls.PKCS1WithSHA384,
+						tls.PKCS1WithSHA512,
+						tls.ECDSAWithSHA1,
+						tls.PKCS1WithSHA1,
+					}},
+					&tls.FakeRecordSizeLimitExtension{0x4001},
+					tls.BoringGREASEECH(),
```

Comment:
I'm pretty sure this was a mistake. Firefox does not use BoringSSL but NSS instead, which uses the same combination of cipher suites but produces different result payload length.

Reply:
just checked the hash, I get the same one using Firefox_120 as the one I get with my real browser Firefox 123.
http://sprunge.us/Px9nxp
Link above contains result of https://tls.peet.ws/api/all from my actual browser

Reply:
Not sure if the hash counts the **payload length** in, but I don't think so.

Reply:
AgAin, NSS and BoringSSL generates almost identical greased ECH with the only difference being the length of the (greased) Encrypted InnerClientHello (aka the payload).